projects/knora-ui/src/lib/viewer/pipes/knoradate.pipe.ts

```diff
@@ -7,25 +7,33 @@ import { KnoraDate } from '@knora/api';
 export class KnoraDatePipe implements PipeTransform {
 
     // TODO: use the optional format parameter and return a string with that format
-    transform(value: KnoraDate, format?: string): string {
-        if (!(value instanceof KnoraDate)) {
+    transform(date: KnoraDate, format?: string): string {
+        if (!(date instanceof KnoraDate)) {
             console.error('Non-KnoraDate provided. Expected a valid KnoraDate');
             return '';
         }
 
-        if (value == null) {
+        if (!date) {
             console.error('Value is null. Expected a valid KnoraDate');
             return '';
         }
 
-        const formattedDate: Date = new Date();
-
-        // js date month is 0-based, so subtract 1
-        formattedDate.setFullYear(value.year, (value.month - 1), value.day);
-
-        const dateStr = formattedDate.toLocaleDateString();
+        // make this fancier
+        switch (format) {
+            case 'dd.MM.YYYY':
+                return `${this.leftPadding(date.day)}.${this.leftPadding(date.month)}.${date.year}`;
+            case 'dd-MM-YYYY':
+                return `${this.leftPadding(date.day)}-${this.leftPadding(date.month)}-${date.year}`;
+            case 'MM/dd/YYYY':
+                return `${this.leftPadding(date.month)}/${this.leftPadding(date.day)}/${date.year}`;
+            default:
+                return `${this.leftPadding(date.day)}.${this.leftPadding(date.month)}.${date.year}`;
+        }
+    }
 
-        return dateStr;
+    // ensures that day and month are always two digits
+    leftPadding(value: number): string {
+        return ('00' + value).slice(-2);
```

Comment:
Wouldn't one 0 be enough?

Reply:
I think this approach is more elegant than what @tobiasschweizer did in https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/blob/f328a5157b33867552ded71a6a3c9a2734e65933/projects/jdnconvertible-calendar-date-adapter/src/lib/jdnconvertible-calendar-date-adapter.ts#L88-L110. Could help him improve the method?

Reply:
you are correct, one 0 is actually enough. Fixed in e52f86d. I'll contact @tobiasschweizer to help him improve his method.